File: 14. Longest Common Prefix/solution.go
func longestCommonPrefix(strs []string) string {
    if(len(strs) == 0) {
        return ""
    }
    if(len(strs) == 1) {
        return strs[0]
    }
    var cur string = strs[0]
    var result string = ""
    for i := 1; i < len(strs); i++ {
        result = ""
        for j := 0; j < min(len(cur), len(strs[i])); j++ {
            if string(cur[j]) == string(strs[i][j]) {
                result += string(strs[i][j])
            } else {
                break
            }
        }
        if result == "" {
            break
        }
        cur = result
    }
    return result
}

func min(a int, b int) int {
    if a >= b {
        return b
    }
    return a
}

File: 14. Longest Common Prefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if(!strs.length) return ""
    if(strs.length === 1) return strs[0]
    let curCommon = strs[0]
    for(let i=1; i<strs.length; i++) {
        let common = ''
        let length = Math.min(curCommon.length, strs[i].length)
        for(let j=0; j<length; j++) {
            if(curCommon[j] === strs[i][j]) {
                common = common + strs[i][j]
            } else {
                break   // !!!
            }
        }
        curCommon = common
        if(!curCommon) return ''
    }
    return curCommon
};